### PR TITLE
feat: Add aws spot instance request

### PR DESCRIPTION
**Pull request by lucastercas**
Add support for `aws_spot_instance_request`

Fixes: #1223

### CODE REVIEW

**Comment by hugorut** [comment]
Thanks for PR this @lucastercas, I'll take a peek at this when I get a moment in the next few days.

File: internal/providers/terraform/aws/testdata/spot_instance_request_test/spot_instance_request_test.golden [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+ Name                                             Monthly Qty  Unit   Monthly Cost 
+                                                                                   
+ aws_spot_instance_request.spot_instance_1                                         
+ └─ Instance usage (Linux/UNIX, spot, m3.medium)          730  hours         $4.89 
+                                                                                   
+ aws_spot_instance_request.spot_instance_2                                         
+ └─ Instance usage (Linux/UNIX, spot, m3.large)           730  hours        $22.85
```
Comment (hugorut) [inline internal/providers/terraform/aws/testdata/spot_instance_request_test/spot_instance_request_test.golden:8]: Sadly this method won't work, as the spot prices change often and will cause the tests to fail. Instead you'll need to test against the price hash. See here: https://github.com/infracost/infracost/blob/59b89e4f850b9f9f1e7930cb2e191549555478c0/internal/providers/terraform/aws/autoscaling_group_test.go#L23 as an example of how to do this.

**Review by hugorut — CHANGES_REQUESTED** [review]
@lucastercas this looks great, thanks. However, we just need to change the way we test the output price. A golden file test won't work unfortunately. Thanks

File: internal/providers/terraform/aws/testdata/spot_instance_request_test/spot_instance_request_test.golden (hunk 1 above)
Comment (lucastercas) [reply internal/providers/terraform/aws/testdata/spot_instance_request_test/spot_instance_request_test.golden:8]: Yeah, no problem. 
I was just waiting for the github pipeline to go back so I can continue, thank you for the help!

Comment (hugorut) [reply internal/providers/terraform/aws/testdata/spot_instance_request_test/spot_instance_request_test.golden:8]: done

File: internal/providers/terraform/aws/testdata/spot_instance_request_test/spot_instance_request_test.golden [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
```
Comment (hugorut) [inline internal/providers/terraform/aws/testdata/spot_instance_request_test/spot_instance_request_test.golden:1]: We can just remove this directory now - given we're using the hash tests

**Review by hugorut — COMMENTED** [review]
Seems like the tests are failing as you just need to add in a check for the `t3_large` spot resource you've got in your tests

File: internal/providers/terraform/aws/testdata/spot_instance_request_test/spot_instance_request_test.tf [hunk 3]
```diff
@@ -0,0 +1,22 @@
+provider "aws" {
```
Comment (hugorut) [inline internal/providers/terraform/aws/testdata/spot_instance_request_test/spot_instance_request_test.tf:1]: this file can be removed

File: internal/providers/terraform/aws/testdata/spot_instance_request_test/spot_instance_request_test.usage.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+version: 0.1
```
Comment (hugorut) [inline internal/providers/terraform/aws/testdata/spot_instance_request_test/spot_instance_request_test.usage.yml:1]: this file can be removed

**Comment by alikhajeh1** [comment]
Any update on this?

**Review by tim775 — APPROVED** [review]

**Review by hugorut — APPROVED** [review]

**Pull request merged**